python/dolfinx/io.py

```diff
@@ -40,6 +41,18 @@ def write_function(self, u, t=0.0, mesh_xpath="/Xdmf/Domain/Grid[@GridType='Unif
         super().write_function(u_cpp, t, mesh_xpath)
 
     def read_mesh(self, name="mesh", xpath="/Xdmf/Domain"):
-        mesh = super().read_mesh(name, xpath)
-        mesh.geometry.coord_mapping = fem.create_coordinate_map(mesh)
+        # Read mesh data from file
+        cell_type, x, cells = super().read_mesh_data(name, xpath)
+
+        # Construct the geometry map
```

Comment:
Could you use here Mesh() method defined below in dolfinx/mesh.py?

Reply:
Not yet - the method in `dolfinx/mesh.py` wraps the deprecated Mesh constructor in the C++ interface.